Makefile

```diff
@@ -672,3 +676,7 @@ nova_deploy_cleanup: ## cleans up the service instance, Does not affect the oper
 	oc kustomize ${DEPLOY_DIR} | oc delete --ignore-not-found=true -f -
 	rm -Rf ${OPERATOR_BASE_DIR}/nova-operator ${DEPLOY_DIR}
 	oc rsh mariadb-openstack mysql -u root --password=${PASSWORD} -ss -e "show databases like 'nova_%';" | xargs -I '{}' oc rsh mariadb-openstack mysql -u root --password=${PASSWORD} -ss -e "drop database {};"
+
+.PHONY: kuttl_keystone
+kuttl_keystone: namespace input deploy_cleanup keystone_deploy_prep keystone
```

Comment:
`deploy_cleanup` fails if it run against a fresh crc deployment:
```
$ make --trace  kuttl_keystone
Makefile:167: target 'namespace' does not exist
bash scripts/gen-namespace.sh
+ '[' -z /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out ']'
+ '[' -z openstack ']'
+ OUT_DIR=/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack
+ '[' '!' -d /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack ']'
+ cat
oc apply -f /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/namespace.yaml
namespace/openstack unchanged
sleep 2
oc project openstack
Already on project "openstack" on server "https://api.crc.testing:6443".
Makefile:184: target 'input' does not exist
bash scripts/gen-input-kustomize.sh openstack osp-secret 12345678
+ NAMESPACE=openstack
+ SECRET=osp-secret
+ PASSWORD=12345678
+ '[' -z openstack ']'
+ '[' -z osp-secret ']'
+ '[' -z 12345678 ']'
+ '[' '!' -d out/openstack/input ']'
+ DIR=out/openstack/input
+ '[' '!' -d out/openstack/input ']'
+ pushd out/openstack/input
~/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/input ~/upstream/git/openstack-k8s-operators/install_yamls
+ cat
oc get secret/osp-secret || oc kustomize /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/input | oc apply -f -
NAME         TYPE     DATA   AGE
osp-secret   Opaque   23     2m50s
Makefile:675: target 'nova_deploy_cleanup' does not exist
oc kustomize /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/nova/cr | oc delete --ignore-not-found=true -f -
error: resource mapping not found for name: "nova" namespace: "openstack" from "STDIN": no matches for kind "Nova" in version "nova.openstack.org/v1beta1"
ensure CRDs are installed first
make: *** [Makefile:676: nova_deploy_cleanup] Error 1
```
So I locally removed that target from the dependency list but then `keystone_deploy_prep` fails
```
$ make --trace  kuttl_keystone
Makefile:167: target 'namespace' does not exist
bash scripts/gen-namespace.sh
+ '[' -z /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out ']'
+ '[' -z openstack ']'
+ OUT_DIR=/home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack
+ '[' '!' -d /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack ']'
+ cat
oc apply -f /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/namespace.yaml
namespace/openstack unchanged
sleep 2
oc project openstack
Already on project "openstack" on server "https://api.crc.testing:6443".
Makefile:184: target 'input' does not exist
bash scripts/gen-input-kustomize.sh openstack osp-secret 12345678
+ NAMESPACE=openstack
+ SECRET=osp-secret
+ PASSWORD=12345678
+ '[' -z openstack ']'
+ '[' -z osp-secret ']'
+ '[' -z 12345678 ']'
+ '[' '!' -d out/openstack/input ']'
+ DIR=out/openstack/input
+ '[' '!' -d out/openstack/input ']'
+ pushd out/openstack/input
~/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/input ~/upstream/git/openstack-k8s-operators/install_yamls
+ cat
oc get secret/osp-secret || oc kustomize /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/input | oc apply -f -
NAME         TYPE     DATA   AGE
osp-secret   Opaque   23     6m13s
Makefile:275: target 'keystone_deploy_cleanup' does not exist
oc kustomize /home/gibi/upstream/git/openstack-k8s-operators/install_yamls/out/openstack/keystone/cr | oc delete --ignore-not-found=true -f -
error: resource mapping not found for name: "keystone" namespace: "openstack" from "STDIN": no matches for kind "KeystoneAPI" in version "keystone.openstack.org/v1beta1"
ensure CRDs are installed first
make: *** [Makefile:276: keystone_deploy_cleanup] Error 1
```
So I guess the intention is to have the openstack-operator itself deployed before we run `kuttl_keystone`? If so, can we add `openstack` as a dependency to this target? Or at least document such dependency somehow. Also similarly we have an implicit dependency between mariadb_deploy and crc_storage target that should be noted.

Reply:
It might be useful to include the https://github.com/openstack-k8s-operators/install_yamls/blob/af41555abe97bff7aac072be9af80aa4887de29a/Makefile#L229 (`openstack_crds`) target as a dependency in this regard